### PR TITLE
First role: ci_make

### PR DESCRIPTION
**Pull request by cjeanner**
This role purpose is to generate indexed playbooks running "make" commands in a specific location.

That way, we will be able to reproduce the various "make" runs in a convenient, ordered way.

### CODE REVIEW

File: ci_framework/roles/ci_make/molecule/default/prepare.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      ansible.builtin.file:
+        path: /tmp/foo
+        state: directory
+    - name: Push dummy Makefile
+      ansible.builtin.file:
+        target: /tmp/foo/Makefile
+        content: |
+          FOO_BAR ?= "toto"
+          help:
+            echo "This is the help thing showing ${FOO_BAR}"
```
Comment (raukadah) [inline ci_framework/roles/ci_make/molecule/default/prepare.yml:31]: - name: push dummy makefile
  ansible.builtin.copy:
      dest:  /tmp/foo/Makefile
       content: |
          FOO_BAR ?= "toto"
          help:
            echo "This is the help thing showing ${FOO_BAR}"

File: ci_framework/roles/ci_make/molecule/default/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+driver:
+  name: podman
+
```
Comment (raukadah) [inline ci_framework/roles/ci_make/molecule/default/molecule.yml:4]: We need to add platforms here:
`platforms:
  - name: centos
    hostname: centos
    pre_build_image: true
    image: ${TRIPLEO_ANSIBLE_MOLECULE_IMAGE:-"ubi9/ubi-init"}
    registry:
      url: ${TRIPLEO_ANSIBLE_MOLECULE_REGISTRY:-"registry.access.redhat.com"}`

File: ci_framework/roles/ci_make/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+# All variables intended for modification should be placed in this file.
+# All variables within this role should have a prefix of "cifmw_ci_make"
+cifmw_ci_make_outputdir: "{{ ansible_user_directory }}/ci_output"
```
Comment (raukadah) [inline ci_framework/roles/ci_make/defaults/main.yml:20]: s/ansible_user_directory/ansible_user_dir

File: ci_framework/roles/ci_make/molecule/default/converge.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# under the License.
+
+
+- name: Converge
+  hosts: all
+  tasks:
+    - name: Default role with chdir
+      vars:
+        chdir: "/tmp/foo"
+      include_role: ci_make
```
Comment (raukadah) [inline ci_framework/roles/ci_make/molecule/default/converge.yml:24]: include_role:
   name: ci_make

File: ci_framework/roles/ci_make/molecule/default/converge.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Role with specific parameter
+      vars:
+        chdir: "/tmp/foo"
+        params:
+          FOO_BAR: "starwars"
+      include_role: ci_make
```
Comment (raukadah) [inline ci_framework/roles/ci_make/molecule/default/converge.yml:30]:
```
include_role:
   name: ci_make
```

**Review by raukadah — COMMENTED** [review]
It is a great start.
Since it is still in draft phase, I will stop adding more comment.

File: ci_framework/roles/ci_make/defaults/main.yml (hunk 3 above)
Comment (cjeanner) [reply ci_framework/roles/ci_make/defaults/main.yml:20]: oh. damn me.

File: ci_framework/roles/ci_make/molecule/default/converge.yml (hunk 4 above)
Comment (cjeanner) [reply ci_framework/roles/ci_make/molecule/default/converge.yml]: actually I had an issue yesterday with my local branch, and didn't see it didn't push ^^'. Sorry!

**Comment by cjeanner** [comment]
Running this: ```make run_ctx_molecule BUILD_VENV_CTX=no MOLECULE_CONFIG=.config/molecule/config_local.yml```

ends with:
```Bash
[...]
PLAY [Prepare] *****************************************************************

TASK [Create lame directory] ***************************************************
changed: [instance]

TASK [Push dummy Makefile] *****************************************************
changed: [instance]

PLAY RECAP *********************************************************************
instance                   : ok=2    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

INFO     Running default > converge

PLAY [Converge] ****************************************************************

TASK [Gathering Facts] *********************************************************
ok: [instance]

TASK [Default role with chdir] *************************************************

TASK [ci_make : Validate we get at least chdir param] **************************
skipping: [instance]

TASK [ci_make : Ensure output directory exists] ********************************
changed: [instance]

TASK [ci_make : Get current amount of playbooks] *******************************
ok: [instance]

TASK [ci_make : Create filename fact] ******************************************
ok: [instance]

TASK [ci_make : Create new make call using index and target] *******************
changed: [instance]

TASK [ci_make : Call make module if wanted] ************************************
changed: [instance]

TASK [ci_make : Expose make output] ********************************************
ok: [instance]

TASK [Validate make output] ****************************************************
ok: [instance] => changed=false 
  msg: All assertions passed

TASK [Role with specific parameter] ********************************************

TASK [ci_make : Validate we get at least chdir param] **************************
skipping: [instance]

TASK [ci_make : Ensure output directory exists] ********************************
ok: [instance]

TASK [ci_make : Get current amount of playbooks] *******************************
ok: [instance]

TASK [ci_make : Create filename fact] ******************************************
ok: [instance]

TASK [ci_make : Create new make call using index and target] *******************
changed: [instance]

TASK [ci_make : Call make module if wanted] ************************************
changed: [instance]

TASK [ci_make : Expose make output] ********************************************
ok: [instance]

TASK [Validate make output] ****************************************************
ok: [instance] => changed=false 
  msg: All assertions passed

PLAY RECAP *********************************************************************
instance                   : ok=15   changed=5    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0
```


I guess this is what one may call a great success :).

**Pull request merged**